Makefile

```diff
@@ -78,7 +78,7 @@ target/wasm32-unknown-unknown/debug/aurora_engine.wasm: Cargo.toml Cargo.lock $(
 etc/eth-contracts/res/EvmErc20.bin: $(shell find etc/eth-contracts/contracts -name "*.sol") etc/eth-contracts/package.json
 	cd etc/eth-contracts && yarn && yarn build
 
-etc/eth-contracts/artifacts/contracts/test/StateTest.sol/StateTest.json: $(shell find etc/eth-contracts/contracts -name "*.sol") etc/eth-contracts/package.json
+etc/eth-contracts/artifacts/contracts/test/StateTest.sol/SelfDestruct.json: $(shell find etc/eth-contracts/contracts -name "*.sol") etc/eth-contracts/package.json
```

Comment:
Out of curiosity, why is self destruct added here (and stateTest.json) removed. Same about other changes in this file with regard with new additions.

Reply:
As it turns out the Makefile was only working by accident previously. The file `StateTest.json` does not exist because the contract in `StateTest.sol` is called `SelfDestruct`. The Rust tests need to have the Solidity test contracts compile for them to work. That compilation did happen previously with only `mainnet-test.wasm` as the `make` requirement because the wasm artifacts depend on `EvmErc20.bin`, and when that file is not kept in the repo then it must be generated by compiling all the Solidity contracts (including the tests). However, with `EvmErc20.bin` committed to the repo, `make` can skip that Solidity compilation step when producing the wasm artifacts (of course this is intentional since we want the engine to be able to compile as a normal Cargo dependency). Therefore we must explicitly say in the Makefile that the Rust tests depend on the Solidity test contract artifacts as well. Of course this dependency was always true it just so happened that it didn't need to be expressed before because of the Solidity compilation triggered earlier in the build process.